utils/codepointtrie/builder/cpp/list_to_ucptrie.cpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#include "unicode/umutablecptrie.h"
+#include "unicode/errorcode.h"
+#include "writesrc.h"
+
+#include <iostream>
+
+/**
+ * list_to_ucptrie: converts a stream of integers to a UCPTrie
+ *
+ * - standard input should be a stream of integers, in order of each code point.
+ * - standard output is the build UCPTrie as a TOML file
+ */
+int main(int argc, char const *argv[]) {
+    if (argc != 4) {
+        std::cerr << "Takes 3 positional arguments: default value, error value, and trie type" << std::endl;
+        return 1;
+    }
+
+    int32_t defaultValue = atoi(argv[1]);
+    int32_t errorValue = atoi(argv[2]);
+
+    UCPTrieType trieType;
+    std::string trieTypeStr(argv[3]);
+    if (trieTypeStr == "small") {
+        trieType = UCPTRIE_TYPE_SMALL;
+    } else if (trieTypeStr == "fast") {
+        trieType = UCPTRIE_TYPE_FAST;
+    } else {
+        std::cerr << "Expected 'small' or 'fast' for 3rd argument" << std::endl;
+        return 1;
+    }
+
+    icu::ErrorCode status;
+
+    icu::LocalUMutableCPTriePointer builder(umutablecptrie_open(
+        defaultValue,
+        errorValue,
+        status));
+    if (status.isFailure()) {
+        std::cerr << status.errorName() << std::endl;
+        return 1;
+    }
+
+    UChar32 cp = 0;
+    std::string line;
+    while (std::getline(std::cin, line)) {
+        int32_t value = atoi(line.data());
+        // std::cerr << cp << " => " << value << std::endl;
+        umutablecptrie_set(builder.getAlias(), cp, value, status);
+        if (status.isFailure()) {
+            std::cerr << status.errorName() << std::endl;
+            return 1;
+        }
+        cp++;
+    }
+    std::cerr << "Saved values for code points up to " << cp << std::endl;
+
+    icu::LocalUCPTriePointer utrie(umutablecptrie_buildImmutable(
+        builder.getAlias(),
+        trieType,
+        UCPTRIE_VALUE_BITS_8, // TODO: Auto-compute
```

Comment:
thought: Auto-compute sounds good, but could be making assumptions that cause problems later on.

Hypothetically, let's say the user wants a CPT for the `Script` property (let's ignore all the `ScriptExtensions` stuff). 
 `Script` is 16-bit wide, but the number of enum values doesn't currently exceed 256.  So we end up assuming `u8` when creating the contents of the CPT here, but the Rust code reading CPTs will be expecting `u16`.

Since we don't & can't know what values the user will provide (since it's an override after all), I think that it's safer and simpler to let the user be clear what value width they are intending.

Reply:
Yes, I wrote this code earlier and forgot to address the TODO. I want to calculate the value width based on the choice of `T`. Will fix.